crates/egui/src/widgets/plot/mod.rs

```diff
@@ -402,9 +402,19 @@ impl Plot {
         self
     }
 
-    /// Whether to allow zooming in the plot. Default: `true`.
-    pub fn allow_zoom(mut self, on: bool) -> Self {
```

Comment:
It would be very easy to maintain backwards compatibility here by keeping these methods and having them set both x and y at the same time.

Question to the reviewers: what's the preference - keep backwards compatibility here or go with a more minimal API?

Reply:
I vote keeping the old one. Not just for backwards compatibility, but also ergonomics: in most cases, you want to control both the same

Reply:
Yeah, good point. It occurs to me now that having it be generic in `T` `where T : Into<AxisBools>` would produce the best of both worlds. That way there's only the one method needed, and it maintains the current ergonomics for setting both at once, while allowing the new flexibility.

The main cost I see there is that it requires exposing AxisBools, which has been private until now. I guess it will also be a little more verbose to disable just one of the axes, as you'll have to do something like `.allow_zoom(AxisBools::new(false, true))` (unless we keep the `allow_zoom_x/y` methods, which doesn't seem worth it). I'll go ahead and update with the new signature, and probably drop the x/y-specific methods.